code/game/machinery/doors/firedoor.dm

```diff
@@ -24,6 +28,8 @@
 	var/nextstate = null
 	var/boltslocked = TRUE
 	var/list/affecting_areas
+	///the list of people trying to open this firelock without any tools
+	var/list/people_trying_to_open = list()
```

Comment:
Make this a lazy list--fire doors are everywhere, and so this'll chew up a good bit of memory.